drivers/pwm/pwm_stm32.c

```diff
@@ -125,43 +125,66 @@ static int get_tim_clk(const struct stm32_pclken *pclken, uint32_t *tim_clk)
 	} else {
 		apb_psc = CONFIG_CLOCK_STM32_D2PPRE2;
 	}
+#else
+	if (pclken->bus == STM32_CLOCK_BUS_APB1) {
+		apb_psc = CONFIG_CLOCK_STM32_APB1_PRESCALER;
+	}
+#if !defined(CONFIG_SOC_SERIES_STM32F0X) && !defined(CONFIG_SOC_SERIES_STM32G0X)
+	else {
+		apb_psc = CONFIG_CLOCK_STM32_APB2_PRESCALER;
+	}
+#endif
+#endif
 
+#if defined(RCC_DCKCFGR_TIMPRE) || defined(RCC_DCKCFGR1_TIMPRE) || \
+	defined(RCC_CFGR_TIMPRE)
 	/*
-	 * Depending on pre-scaler selection (TIMPRE), timer clock frequency
-	 * is defined as follows:
+	 * There are certain series (some F4, F7 and H7) that have the TIMPRE bit
+	 * to control the clock frequency of all the timers connected to APB1 and
+	 * APB2 domains. Depending on pre-scaler selection (TIMPRE), timer clock
+	 * frequency can be calculated as follows:
+	 *
+	 * - TIMPRE=0: If the APB{1,2} prescaler is configured to a division
+	 *   factor of 1 or 2 then the timer clock equals to HCLK. Otherwise the
```

Comment:
This comment is definitely welcome. Thought, I could understand what it means only after I had a check to the code :-). I think it would deserve some rework.
Let me have a try:
```
Up to a certain threshold value of APB{1,2} prescaler,  timer clock equals to HCLK. This threshold value depends on TIMPRE setting (2 if TIMPRE=0, 4 if TIMPRE=1).
Above threshold, timer clock is set to a multiple of HCLK ((2 if TIMPRE=0, 4 if TIMPRE=1).
```
Is that anyhow clearer ?

Reply:
@erwango Much better. Thanks. You should probably suggest to put that into new datasheet revisions :laughing: